lib/client/profile.go

```diff
@@ -133,7 +133,11 @@ func (h *profiles) convertKVPairToAPI(d *model.KVPair) (unversioned.Resource, er
 	ap.Metadata.Labels = bp.Labels
 	ap.Spec.IngressRules = rulesBackendToAPI(bp.Rules.InboundRules)
 	ap.Spec.EgressRules = rulesBackendToAPI(bp.Rules.OutboundRules)
-	ap.Spec.Tags = bp.Tags
+	if len(bp.Tags) == 0 {
```

Comment:
This is required because of the "opposite" code in compat.go::toTagsLabelsRules.

IIRC that code was required because the Python version of Felix didn't handle nil values of labels and tags.  Once we move over to golang Felix we should be able to get rid of the conversion code I assume.